src/main/java/com/istb/app/controller/VerificationController.java

```diff
@@ -0,0 +1,34 @@
+package com.istb.app.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import com.istb.app.entities.Profesor;
+import com.istb.app.entities.Usuario;
+import com.istb.app.services.profesor.ProfesorService;
+import com.istb.app.services.user.UserService;
+
+@Controller
+public class VerificationController {
+
+	@Autowired
+	ProfesorService profesorService;
+
+	@Autowired
+	UserService userService;
+	
+	@GetMapping("/verify/{id}/{token}")
+	public String verify(@PathVariable int id, @PathVariable String token) {
+		Profesor profesor = profesorService.findById(id);
+		if (profesor.getTokenVerification().equals(token)) {
+			Usuario usuario = userService.findById(profesor.getUsuario().getId());
```

Comment:
Para evitar la consulta utiliza el usuario de la instancia *profesor*
```suggestion
			Usuario usuario = profesor.getUsuario();
```